src/renderer/components/deposit/add/SymDeposit.tsx

```diff
@@ -686,7 +688,20 @@ export const SymDeposit: React.FC<Props> = (props) => {
       // check approved status
       checkApprovedStatus()
     }
-  }, [checkApprovedStatus, oPoolRouter, resetApproveState, resetIsApprovedState])
+
+    if (!eqOAsset.equals(prevAsset.current, O.some(asset))) {
```

Comment:
All of following logic after this guard ^ is hard to understand. Pls clarify it here and add comments. For example: I don't understand, why `prevAsset.current` is not set, whenever `prevAsset` and `asset` are different ...